lib/model/autocomplete.dart

```diff
@@ -289,6 +538,38 @@ class MentionAutocompleteView extends ChangeNotifier {
         streamId: streamId, senderId: userB.userId));
   }
 
+  @override
+  Iterable<Object> getSortedItemsToTest(MentionAutocompleteQuery query) {
+    switch (query) {
+      case UserMentionAutocompleteQuery():
+        return sortedUsers;
+      case ChannelMentionAutocompleteQuery(:var raw):
+        final matches = store.streams.values.where((item) => item.name.contains(raw));
+        final nameMatches = _triage(query: raw, objects: matches, getItem: (s) => s.name,
+          sortingComparator: _channelComparator);
+        final descriptionMatches = _triage(query: raw, objects: nameMatches.rest, getItem: (s) => s.description,
+          sortingComparator: _channelComparator);
+        return [...nameMatches.matches, ...descriptionMatches.matches, ...descriptionMatches.rest];
+    }
+  }
+
+  @override
+  MentionAutocompleteResult? testItem(MentionAutocompleteQuery query, Object item) {
+    switch (query) {
+      case UserMentionAutocompleteQuery():
+        item as User;
```

Comment:
The need for this cast here (and the corresponding fact that `CandidateT` is just `Object` when the truth is that the candidate type is determined by the query type and is always either User or ZulipStream) is also a sign that this isn't the right arrangement of the types.